Akka HTTP Client Connection-level (1.0-RC3) fails the documentation code example

The code given in the documentation [here](http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0-RC3/scala/http/client-side/connection-level.html) throws an exception:

```scala
import scala.concurrent.Future
import akka.stream.ActorFlowMaterializer
import akka.stream.scaladsl._
import akka.http.scaladsl.model._
import akka.http.scaladsl.Http
 
implicit val system = ActorSystem()
implicit val materializer = ActorFlowMaterializer()
 
val connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]] =
  Http().outgoingConnection("http://akka.io")
val responseFuture: Future[HttpResponse] =
  Source.single(HttpRequest(uri = "/"))
    .via(connectionFlow)
    .runWith(Sink.head)
```

The exception details and trace are as follows:

```
[ERROR] [05/28/2015 23:34:30.669] [default-akka.actor.default-dispatcher-3] [ActorSystem(default)] Outgoing request stream error
akka.http.scaladsl.model.IllegalUriException: Illegal URI host: Invalid input ':', expected lower-reg-name-char, UPPER_ALPHA, '%' or 'EOI' (line 1, column 5): http://akka.io
    ^
	at akka.http.scaladsl.model.IllegalUriException$.apply(ErrorInfo.scala:40)
	at akka.http.scaladsl.model.Uri$.fail(Uri.scala:748)
	at akka.http.impl.model.parser.UriParser.fail(UriParser.scala:61)
	at akka.http.impl.model.parser.UriParser.parseHost(UriParser.scala:51)
	at akka.http.scaladsl.model.Uri$Host$.apply(Uri.scala:350)
	at akka.http.scaladsl.model.headers.Host$.apply(headers.scala:95)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.renderHeaders$1(HttpRequestRendererFactory.scala:110)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.onPush(HttpRequestRendererFactory.scala:139)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.onPush(HttpRequestRendererFactory.scala:42)
	at akka.stream.impl.fusing.OneBoundedInterpreter$$anon$1.run(Interpreter.scala:436)
	at akka.stream.impl.fusing.OneBoundedInterpreter$State$class.progress(Interpreter.scala:245)
	at akka.stream.impl.fusing.OneBoundedInterpreter$$anon$1.progress(Interpreter.scala:434)
	at akka.stream.impl.fusing.OneBoundedInterpreter.akka$stream$impl$fusing$OneBoundedInterpreter$$execute(Interpreter.scala:580)
	at akka.stream.impl.fusing.OneBoundedInterpreter$State$class.execute(Interpreter.scala:241)
	at akka.stream.impl.fusing.OneBoundedInterpreter$EntryState.execute(Interpreter.scala:666)
	at akka.stream.stage.AbstractStage.enterAndPush(Stage.scala:64)
	at akka.stream.impl.fusing.BatchingActorInputBoundary$$anonfun$upstreamRunning$1.applyOrElse(ActorInterpreter.scala:143)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at akka.stream.impl.SubReceive.apply(Transfer.scala:16)
	at akka.stream.impl.SubReceive.apply(Transfer.scala:12)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at akka.stream.impl.SubReceive.applyOrElse(Transfer.scala:12)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:170)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:467)
	at akka.stream.impl.fusing.ActorInterpreter.aroundReceive(ActorInterpreter.scala:352)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.ActorCell.invoke(ActorCell.scala:487)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
	at akka.dispatch.Mailbox.run(Mailbox.scala:220)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

--- a/akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala
+++ b/akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala
@@ -22,7 +22,7 @@ class HttpClientExampleSpec extends WordSpec with Matchers {
     implicit val materializer = ActorFlowMaterializer()
 
     val connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]] =
-      Http().outgoingConnection("http://akka.io")
+      Http().outgoingConnection("akka.io")
     val responseFuture: Future[HttpResponse] =
       Source.single(HttpRequest(uri = "/"))
         .via(connectionFlow)
@@ -44,7 +44,7 @@ class HttpClientExampleSpec extends WordSpec with Matchers {
     implicit val materializer = ActorFlowMaterializer()
 
     // construct a pool client flow with context type `Int`
-    val poolClientFlow = Http().cachedHostConnectionPool[Int]("http://akka.io")
+    val poolClientFlow = Http().cachedHostConnectionPool[Int]("akka.io")
     val responseFuture: Future[(Try[HttpResponse], Int)] =
       Source.single(HttpRequest(uri = "/") -> 42)
         .via(poolClientFlow)

--- a/akka-http-core/src/test/scala/akka/http/impl/engine/client/LowLevelOutgoingConnectionSpec.scala
+++ b/akka-http-core/src/test/scala/akka/http/impl/engine/client/LowLevelOutgoingConnectionSpec.scala
@@ -350,7 +350,6 @@ class LowLevelOutgoingConnectionSpec extends AkkaSpec("akka.loggers = []\n akka.
   class TestSetup {
     val requests = TestPublisher.manualProbe[HttpRequest]
     val responses = TestSubscriber.manualProbe[HttpResponse]
-    val remoteAddress = new InetSocketAddress("example.com", 0)
 
     def settings = ClientConnectionSettings(system)
       .copy(userAgentHeader = Some(`User-Agent`(List(ProductVersion("akka-http", "test")))))
@@ -359,7 +358,7 @@ class LowLevelOutgoingConnectionSpec extends AkkaSpec("akka.loggers = []\n akka.
       val netOut = TestSubscriber.manualProbe[ByteString]
       val netIn = TestPublisher.manualProbe[ByteString]
 
-      FlowGraph.closed(OutgoingConnectionBlueprint(Host(remoteAddress), settings, NoLogging)) { implicit b ⇒
+      FlowGraph.closed(OutgoingConnectionBlueprint(Host("example.com"), settings, NoLogging)) { implicit b ⇒
         client ⇒
           import FlowGraph.Implicits._
           Source(netIn) ~> Flow[ByteString].map(SessionBytes(null, _)) ~> client.in2

